tests/ieee802154_hal/init_devs.c

```diff
@@ -87,4 +105,16 @@ void ieee802154_hal_test_init_devs(ieee802154_dev_cb_t cb, void *opaque)
         socket_zep_setup(&_socket_zeps[0], &socket_zep_params[0]);
     }
 #endif
+
+#ifdef MODULE_MRF24J40
+    auto_init_event_thread();
```

Comment:
This will end up calling `thread_create()` multiple times if there are two drivers that use the bhp

Reply:
good catch. Addressed in https://github.com/RIOT-OS/RIOT/pull/18472/commits/6b102f4c51c72d6471cb64cd816e27a134c7aec0

Reply:
Why so complicated?

```patch
--- a/tests/ieee802154_hal/init_devs.c
+++ b/tests/ieee802154_hal/init_devs.c
@@ -50,6 +50,10 @@ static bhp_event_t kw2xrf_bhp[KW2XRF_NUM];
 
 void ieee802154_hal_test_init_devs(ieee802154_dev_cb_t cb, void *opaque)
 {
+    if (IS_USED(MODULE_EVENT_THREAD)) {
+        auto_init_event_thread();
+    }
+
     /* Call the init function of the device (this should be handled by
      * `auto_init`) */
     ieee802154_dev_t *radio = NULL;
```

Reply:
Sure, I can do that.